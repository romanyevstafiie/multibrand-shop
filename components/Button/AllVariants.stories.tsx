import React from 'react';
import Container from '../Container/Container';
import { LoadingButton, Primary } from './Button.stories';
export default {
  title: 'test/AllVariants',
  decorators: [(story) => <Container>{story()}</Container>],
};

export const AllVariants = (): JSX.Element => {
  return (
    <>
      <Primary
        text="hello"
        disabled={false}
        isLoading={false}
        isCompleted={false}
      />
      <LoadingButton disabled={false} isLoading={true} isCompleted={false} />
    </>
  );
};
