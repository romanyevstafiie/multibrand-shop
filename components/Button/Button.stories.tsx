import { Story } from '@storybook/react';
import React from 'react';
import Container from '../Container/Container';
import TestButton, { Props } from './Button';

export default {
  title: 'test/FlexibleButton',
  component: TestButton,
  // args: {
  //   children: 'Button',
  // },
  decorators: [(story) => <Container>{story()}</Container>],
};

const Template: Story<Props> = (args) => <TestButton {...args} />;

export const Primary = Template.bind({});
Primary.args = { children: 'Primary' };
export const LoadingButton = Template.bind({});
LoadingButton.args = { isLoading: true, text: 'Loading' };
export const DisabledButton = Template.bind({});
DisabledButton.args = { disabled: true, text: 'Disabled' };
export const CompletedButton = Template.bind({});
CompletedButton.args = { isCompleted: true, text: 'Completed ü•≥' };

// extend args
export const ExtendArgs = Template.bind({});
ExtendArgs.args = {
  ...Primary.args,
  text: '‚≠ê',
};
