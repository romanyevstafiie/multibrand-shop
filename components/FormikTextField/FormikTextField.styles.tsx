import styled, { css } from 'styled-components';

export const Container = styled.fieldset`
  position: relative;
  padding: 0.25rem 0;
  border: 0;
  width: 100%;
`;

const labelWithValueStyles = css`
  top: -11px;
  left: 0;
  font-size: 12px;
  color: ${({ theme }) => theme.primaryGolden};
`;

interface LabelProps {
  hasValue: boolean;
}

export const Label = styled.label<LabelProps>`
  position: absolute;
  top: 50%;
  left: 0.5rem;
  transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  -webkit-transform: translateY(-50%);
  cursor: text;
  user-select: none;
  transition: 0.15s ease-in-out;
  color: ${({ theme }) => theme.lightGray};
  text-transform: capitalize;

  // if label is not empty the below styles will override some default styles, thereby positioning label above input
  ${({ hasValue }) => hasValue && labelWithValueStyles};
`;

interface InputProps {
  hasError: boolean;
}

export const Input = styled.input<InputProps>`
  display: block;
  height: 3rem;
  padding: 0.5rem;
  width: 100%;
  color: ${({ theme }) => theme.primaryWhite};
  background: ${({ theme }) => theme.dashboardGrayCard};
  border: 1px solid;
  border-color: ${({ theme, hasError }) =>
    hasError ? theme.primaryRed : theme.borderGrayColor};
  border-radius: ${({ theme }) => theme.borderRadius};

  &:focus-visible {
    outline: none;
    box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
    border-color: ${({ theme }) => theme.primaryGolden};
  }
`;

interface ErrorProps {
  hasError: boolean;
}

export const Error = styled.div<ErrorProps>`
  visibility: ${({ hasError }) => (hasError ? 'visible' : 'hidden')};
  position: absolute;
  display: flex;
  align-items: center;
  margin-top: 1px;
  color: ${({ theme }) => theme.primaryRed};
  font-size: 12px;

  .icon {
    font-size: 12px;
    margin-right: 5px;
    color: ${({ theme }) => theme.primaryRed};
  }
`;
