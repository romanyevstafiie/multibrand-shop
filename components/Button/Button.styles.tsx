import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

/***
 * DEFAULT BUTTON STATE
 */
const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;
/***
 * HOVERED BUTTON STATE
 */
const hoveredButton = css`
  opacity: 0.95;
  transform: translateY(-2px);
  transition: transform 250ms cubic-bezier(0.3, 0.7, 0.4, 1.5);
`;
/***
 * FOCUSED BUTTON STATE
 */
const focusedButton = css`
  &:focus-visible {
    box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryBg};
  }
`;
/***
 * DISABLED BUTTON STATE
 */
const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;
/***
 * ACTIVE BUTTON STATE
 */
const activeButton = css`
  transform: scale(0.97);

  :before {
    display: none;
  }

  :after {
    content: '';
    position: absolute;
    z-index: 1;
    bottom: 1px;
    left: 1px;
    right: 1px;
    height: 55%;
    background: linear-gradient(
      rgba(255, 255, 255, 0),
      rgba(255, 255, 255, 0.75)
    );
    border-radius: 14px 14px 14px 14px/ 4px 4px 100px 100px;
  }
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;
/***
 * LOADING BUTTON STATE
 */
const movingGradient = css`repeating-linear-gradient(60deg,
#ffa500,
#ffa500 10px,
${({ theme }) => theme.primaryGolden} 10px,
${({ theme }) => theme.primaryGolden} 20px)
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.secondaryBg} ${movingGradient};
    animation: ${loadingKeyframes} 2s infinite linear;
  }

  & > span {
    opacity: 0.5;
  }

  cursor: wait;
`;
/***
 * COMPLETED BUTTON STATEAir
 */
const completedButton = css`
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

/***
 * BUTTON BASE STYLES
 */
export const BTN = styled.button<Props>`
  position: relative;
  overflow: hidden;

  display: inline-flex;
  align-items: center;
  justify-content: center;

  padding: 0.75rem 1rem;

  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 600;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) => theme.primaryWhite};
  cursor: pointer;
  outline: none;

  text-shadow: rgb(0 0 0 / 15%) 1px 1px 1px;
  transition: 0.3s all ease;

  :before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 1px;
    left: 2px;
    right: 2px;
    height: 35%;
    background: linear-gradient(rgba(255, 255, 255, 1), rgba(255, 255, 255, 0));
    border-radius: 4px 4px 100px 100px / 14px 14px 14px 14px;
  }

  ${(p) => !p.isLoading && defaultButton};

  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }

  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }

  &:active {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && activeButton};
  }

  ${(p) => p.isLoading && loadingButton};

  &:disabled {
    ${disabledButton};
  }

  ${(p) => p.isCompleted && completedButton};

  @media ${({ theme }) => theme.media.phone} {
    font-size: 0.75rem;
  }
`;
