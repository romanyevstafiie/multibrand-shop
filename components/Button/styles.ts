import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
