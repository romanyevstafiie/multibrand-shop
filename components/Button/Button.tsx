import React, { FC, ReactNode } from 'react';
import { StyledButton } from './styles';

export interface Props {
  isLoading?: boolean;
  isCompleted?: boolean;
  disabled?: boolean;
  text?: string;
  children?: ReactNode;
}

/**
 * Reusable generic button
 *@function Button
 *@param {boolean} isLoading - set styles for loading state
 *@param {boolean} isCompleted - set styles for completed state
 *@param {boolean} disabled - set styles for disabled state
 *@param {string} text - set button text
 *@param {children} ReactNode - set button children
 *@returns {JSX.Element} - rendered button
 */
const Button: FC<
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > &
    Props
> = ({ children, isLoading, isCompleted, text, ...rest }: Props) => {
  return (
    <StyledButton
      isLoading={isLoading}
      isCompleted={isCompleted}
      {...rest}
      className="btn"
    >
      {children}
      {text && text}
    </StyledButton>
  );
};

export default Button;
