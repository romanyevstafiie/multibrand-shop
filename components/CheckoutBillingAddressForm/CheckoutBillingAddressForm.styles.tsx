import styled from 'styled-components';
import { slideInTop } from '../../styles/reusableStyles';

export const Container = styled.section`
  animation: ${slideInTop} 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;

  form {
    display: flex;
    flex-direction: column;
    gap: 1.75rem;

    input {
      background-color: ${({ theme }) => theme.dashboardGrayBg};
    }

    div {
      margin: 0;
      padding: 0;
    }
  }
`;

export const FormControl = styled.fieldset`
  display: flex;
  gap: 1.75rem;
  outline: 0;
  border: 0;

  @media ${({ theme }) => theme.media.tablet} {
    flex-direction: column;
  }
`;
