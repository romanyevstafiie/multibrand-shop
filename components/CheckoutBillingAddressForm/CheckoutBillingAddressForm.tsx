import React from 'react';
import { toast } from 'react-toastify';
import { useActions } from '../../hooks/useActions';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import { UserShippingInfo } from '../../types';
import * as Styled from './CheckoutBillingAddressForm.styles';
import * as yup from 'yup';
import { Form, Formik } from 'formik';
import FormikTextField from '../FormikTextField/FormikTextField';
import Button from '../Button/Button';

/**
 *Renders form to collect user shipping information
 *@function CheckoutShippingInfo
 *@returns {JSX.Element} - Rendered CheckoutBillingAddressForm component
 */

const CheckoutBillingAddressForm = (): JSX.Element => {
  const { isLoading, isShippingInfoCollected } = useTypedSelector(
    (state) => state.checkout
  );
  const { collectCustomerShippingInfo } = useActions();

  return (
    <Styled.Container>
      <Formik
        initialValues={{
          firstName: '',
          lastName: '',
          email: '',
          address: '',
          city: '',
          country: '',
          state: '',
          zipCode: '',
          phone: '',
        }}
        validationSchema={validationSchema}
        onSubmit={(data) => {
          // connect all data into userShippingInformation
          const userShippingInformation: UserShippingInfo = {
            firstName: data.firstName,
            lastName: data.lastName,
            email: data.email,
            city: data.city,
            address: data.address,
            country: data.country,
            state: data.state,
            zipCode: data.zipCode,
            phone: data.phone,
          };

          // dispatch action to redux
          collectCustomerShippingInfo(userShippingInformation);

          // if everything is correct, proceed to shipping stage
          toast.success('ðŸŽ‰ Shipping information is saved ðŸŽ‰');
        }}
      >
        <Form>
          <Styled.FormControl>
            {/* FIRST AND LAST NAME */}
            <FormikTextField
              type="text"
              name="firstName"
              placeholder="First Name"
            />
            <FormikTextField
              type="text"
              name="lastName"
              placeholder="Last Name"
            />
          </Styled.FormControl>
          {/*EMAIL*/}
          <FormikTextField type="email" name="email" placeholder="Email" />
          {/* ADDRESS */}
          <FormikTextField type="text" name="address" placeholder="Address" />
          {/* CITY */}
          <FormikTextField type="text" name="city" placeholder="City" />
          {/* COUNTRY, STATE AND ZIP-CODE */}
          <Styled.FormControl>
            <FormikTextField type="text" name="country" placeholder="Country" />
            <FormikTextField type="text" name="state" placeholder="State" />
            <FormikTextField
              type="text"
              name="zipCode"
              placeholder="ZIP Code"
            />
          </Styled.FormControl>
          {/* PHONE */}
          <FormikTextField type="tel" name="phone" placeholder="Phone" />

          <Button
            text={
              isShippingInfoCollected
                ? 'Billing Address Was Updated'
                : 'Update Billing Address'
            }
            isCompleted={isShippingInfoCollected}
            isLoading={isLoading}
            type="submit"
          />
        </Form>
      </Formik>
    </Styled.Container>
  );
};

const validationSchema = yup.object({
  firstName: yup
    .string()
    .required('Please add first name')
    .max(25, 'First name cannot exceed 25 symbols'),

  lastName: yup
    .string()
    .required('Please add last name')
    .max(25, 'Last name cannot exceed 25 symbols'),

  email: yup
    .string()
    .email('Email is incorrect')
    .required('Please add email')
    .max(35, 'Email cannot exceed 35 symbols'),

  address: yup
    .string()
    .required('Please add address')
    .max(50, 'Address cannot exceed 50 symbols'),

  city: yup
    .string()
    .required('Please add city')
    .max(25, 'City cannot exceed 25 symbols'),

  country: yup
    .string()
    .required('Please add country')
    .max(25, 'Country cannot exceed 25 symbols'),

  state: yup
    .string()
    .required('Please add state')
    .max(20, 'State cannot exceed 20 symbols'),

  zipCode: yup
    .string()
    .required('Please add zip code')
    .max(15, 'Zip code cannot exceed 15 symbols'),

  phone: yup
    .string()
    .required('Please add phone number')
    .max(25, 'Phone number cannot exceed 15 symbols'),
});

export default CheckoutBillingAddressForm;
