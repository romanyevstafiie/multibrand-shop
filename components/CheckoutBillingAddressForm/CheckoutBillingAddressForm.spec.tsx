import { mount } from 'enzyme';
import React from 'react';
import { withReduxAndStyledProviders } from '../../test/testUtils';
import CheckoutBillingAddressForm from './CheckoutBillingAddressForm';
import FormikTextField from '../FormikTextField/FormikTextField';
import renderer from 'react-test-renderer';
import Button from '../Button/Button';

/**
 * Setup function for the component
 * @returns {JSX.Element} mountWrapper
 */
const setup = () => {
  return mount(withReduxAndStyledProviders(<CheckoutBillingAddressForm />));
};

describe('Checkout Billing Address form component', () => {
  const wrap = setup();

  it('should render without throwing an error', () => {
    expect(wrap);
    expect(wrap.length).toBe(1);
  });

  it('should render all form controls', () => {
    const formControl = wrap.find(FormikTextField);
    expect(formControl.length).toBe(9);
  });
  it('should render submit button', () => {
    const submitBtn = wrap.find(Button);
    expect(submitBtn.length).toBe(1);
  });

  test('it applies default styles and matches snapshot', () => {
    const tree = renderer
      .create(withReduxAndStyledProviders(<CheckoutBillingAddressForm />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
