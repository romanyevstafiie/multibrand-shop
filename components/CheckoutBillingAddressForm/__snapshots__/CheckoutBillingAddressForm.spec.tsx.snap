// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Billing Address form component it applies default styles and matches snapshot 1`] = `
<section
  className="CheckoutBillingAddressFormstyles__Container-sc-ml5ma3-0 czUgzW"
>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <fieldset
      className="CheckoutBillingAddressFormstyles__FormControl-sc-ml5ma3-1 jSvROU"
    >
      <fieldset
        className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
      >
        <input
          className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
          name="firstName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
          htmlFor="firstName"
        >
          First Name
        </label>
        <div
          className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
          
        </div>
      </fieldset>
      <fieldset
        className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
      >
        <input
          className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
          name="lastName"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
          htmlFor="lastName"
        >
          Last Name
        </label>
        <div
          className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
          
        </div>
      </fieldset>
    </fieldset>
    <fieldset
      className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
    >
      <input
        className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
      <label
        className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
        htmlFor="email"
      >
        Email
      </label>
      <div
        className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
          />
        </svg>
        
      </div>
    </fieldset>
    <fieldset
      className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
    >
      <input
        className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
        htmlFor="address"
      >
        Address
      </label>
      <div
        className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
          />
        </svg>
        
      </div>
    </fieldset>
    <fieldset
      className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
    >
      <input
        className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
        htmlFor="city"
      >
        City
      </label>
      <div
        className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
          />
        </svg>
        
      </div>
    </fieldset>
    <fieldset
      className="CheckoutBillingAddressFormstyles__FormControl-sc-ml5ma3-1 jSvROU"
    >
      <fieldset
        className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
      >
        <input
          className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
          name="country"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
          htmlFor="country"
        >
          Country
        </label>
        <div
          className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
          
        </div>
      </fieldset>
      <fieldset
        className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
      >
        <input
          className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
          name="state"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
          htmlFor="state"
        >
          State
        </label>
        <div
          className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
          
        </div>
      </fieldset>
      <fieldset
        className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
      >
        <input
          className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
          name="zipCode"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
          htmlFor="zipCode"
        >
          ZIP Code
        </label>
        <div
          className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
        >
          <svg
            className="icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
            />
          </svg>
          
        </div>
      </fieldset>
    </fieldset>
    <fieldset
      className="FormikTextFieldstyles__Container-sc-j5e6g8-0 gfHFLG"
    >
      <input
        className="FormikTextFieldstyles__Input-sc-j5e6g8-2 hLUiqp"
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        type="tel"
        value=""
      />
      <label
        className="FormikTextFieldstyles__Label-sc-j5e6g8-1 fMQMhn"
        htmlFor="phone"
      >
        Phone
      </label>
      <div
        className="FormikTextFieldstyles__Error-sc-j5e6g8-3 frNgMi"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
          />
        </svg>
        
      </div>
    </fieldset>
    <button
      className="styles__StyledButton-sc-fh9flx-0 zUcJK btn"
      type="submit"
    >
      Update Billing Address
    </button>
  </form>
</section>
`;
